DragonSlayer.cpp

```diff
@@ -10,12 +10,16 @@ void DragonSlayer::attack(Character& other)
     std::cout << name << " is attacking " << other.getName() << " !!" << std::endl;
     if( auto* dragon = dynamic_cast<Dragon*>(&other) )
     {
-        assert(false);
+
         //DragonSlayers get a 10x boost when attacking dragons, from their attack item.
         //so they should USE their attack item before attacking the dragon... 
         //note: they should only use the item if the dragon's hitpoints are > 0...
         //note: items are single-use only, so you need to reset it after use.  
         //look in the Character class for how the other item types are reset after use.
+
+        attackItem->use(this);
+        attackItem.reset();
```

Comment:
What do we always do before using a pointer (`unique_ptr`)? 
also:
```
note: they should only use the item if the dragon's hitpoints are > 0...
```
- [ ] revise

Reply:
always check if a pointer is nullptr first before using it!